call.go

```diff
@@ -29,14 +29,27 @@ import (
 func (cc *ClientConn) Invoke(ctx context.Context, method string, args, reply interface{}, opts ...CallOption) error {
 	// allow interceptor to see all applicable call options, which means those
 	// configured as defaults from dial option as well as per-call options
-	opts = append(cc.dopts.callOptions, opts...)
+	opts = combine(cc.dopts.callOptions, opts)
```

Comment:
Just brainstorming here; Do these options really need to be combined? Could we just apply the default first followed by non defaults or something? I say that because allocating and moving memory(albeit small) seems like an overhead in critical path.

Reply:
> Do these options really need to be combined?

They all should get passed to the interceptor, which means they all need to be together unless we change the interceptor API.